### PR TITLE
ReviewDB: Add Review Modal & Pagination

### PR DESCRIPTION
**Pull request by mantikafasi**
probably somethimh

### CODE REVIEW

**Comment by mantikafasi** [comment]
prohbably memregable and breedeble with vencord

File: src/plugins/reviewDB/index.tsx [hunk 1]
```diff
@@ -18,17 +18,20 @@
...
 import { Settings } from "@api/Settings";
 import ErrorBoundary from "@components/ErrorBoundary";
+import { OpenExternalIcon } from "@components/Icons";
 import { Devs } from "@utils/constants";
 import definePlugin, { OptionType } from "@utils/types";
-import { Alerts, Button } from "@webpack/common";
-import { User } from "discord-types/general";
+import { Alerts, Button, Menu, Text, Tooltip, useState } from "@webpack/common";
+import { Guild, User } from "discord-types/general";
+import { cl } from "plugins/permissionsViewer/utils";
```
Comment (Vendicated) [inline src/plugins/reviewDB/index.tsx:29]: bad import

File: src/plugins/reviewDB/index.tsx [hunk 2]
```diff
@@ -75,7 +78,12 @@ export default definePlugin({
             description: "ReviewDB website",
             component: () => (
                 <Button onClick={() => {
-                    window.open("https://reviewdb.mantikafasi.dev");
+                    if (Settings.plugins.ReviewDB.token) {
+                        window.open("https://reviewdb.mantikafasi.dev/api/redirect?token=" + encodeURIComponent(Settings.plugins.ReviewDB.token));
```
Comment (Vendicated) [inline src/plugins/reviewDB/index.tsx:82]: this should use VencordNative instead

File: src/plugins/reviewDB/index.tsx [hunk 3]
```diff
@@ -106,6 +115,17 @@ export default definePlugin({
                     showToast("You have new reviews on your profile!");
             }
 
+            addContextMenuPatch("guild-header-popout", (children, props: { guild: Guild, onClose(): void; }) => () => {
```
Comment (Vendicated) [inline src/plugins/reviewDB/index.tsx:118]: remove this in patch in stop

File: src/plugins/reviewDB/index.tsx [hunk 4]
```diff
@@ -136,9 +159,75 @@ export default definePlugin({
...
+                            marginBottom: "8px",
+                            color: "var(--header-primary)",
+                            display: "inline"
+                        }}
+                    >
+                        User Reviews
+                    </Text>
+
+                    <div>
+                        <Tooltip text="Open Reviews Modal">
```
Comment (Vendicated) [inline src/plugins/reviewDB/index.tsx:185]:
```suggestion
                        <Tooltip text="Open Review Modal">
```

File: src/plugins/reviewDB/index.tsx [hunk 5]
```diff
@@ -136,9 +159,75 @@ export default definePlugin({
...
+                            {tooltipProps => (
+                                <button
+                                    {...tooltipProps}
+                                    className={cl("userperms-permdetails-btn")}
```
Comment (Vendicated) [inline src/plugins/reviewDB/index.tsx:189]: oh god i see now, you're literally using permissionviewer's css classes. You should instead extract this code into a component in @components and refactor both plugins to use it!

Comment (mantikafasi) [reply src/plugins/reviewDB/index.tsx:189]: who is that guy

File: src/plugins/reviewDB/index.tsx (hunk 3 above)
Comment (mantikafasi) [reply src/plugins/reviewDB/index.tsx:118]: yop

File: src/plugins/reviewDB/index.tsx (hunk 2 above)
Comment (mantikafasi) [reply src/plugins/reviewDB/index.tsx:82]: agree

File: src/components/ExpandableHeader.tsx [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+import { classNameFactory } from "@api/Styles";
+import { Text, Tooltip, useState } from "@webpack/common";
+export const cl = classNameFactory("vc-expandableheader-");
+import "./ExpandableHeader.css";
+
+export interface ExpandableHeaderProps {
+    onMeatballClick?: () => void;
```
Comment (Vendicated) [inline src/components/ExpandableHeader.tsx:25]: MEATBALL

File: src/components/ExpandableHeader.tsx [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+                        color: "var(--header-primary)",
+                        display: "inline"
+                    }}
+                >
+                    {headerText}
+                </Text>
+
+                <div className={cl("center-flex")}>
+                    {
+                        buttons?.map((button, i) => (
```
Comment (Vendicated) [inline src/components/ExpandableHeader.tsx:58]: what? this map is pointless

File: src/components/ExpandableHeader.tsx (hunk 6 above)
Comment (mantikafasi) [reply src/components/ExpandableHeader.tsx:25]: IDK WHAT ELSE TO CALL IT

File: src/components/ExpandableHeader.tsx (hunk 7 above)
Comment (mantikafasi) [reply src/components/ExpandableHeader.tsx:58]: wait can I just put {buttons}

File: src/components/ExpandableHeader.tsx (hunk 6 above)
Comment (TheKodeToad) [reply src/components/ExpandableHeader.tsx:25]: call it "burrito"

File: src/components/ExpandableHeader.tsx (hunk 7 above)
Comment (Vendicated) [reply src/components/ExpandableHeader.tsx:58]: yes lmao

but undefined isn't a valid component so do `{buttons ?? null}`

**Comment by mantikafasi** [comment]
breed

**Comment by somerandomcloud** [comment]
me and your mother

**Pull request merged**